### PR TITLE
feat: support functions nvl and nvl2

### LINKED ISSUE #14381: Feature:  support function nvl aliase ifnull
**Summary**

Oracle use nvl as ifnull.
We just support ifnull as nvl.

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

- Rewrite nvl(x, y) to if(is_not_null(x), x, y), nvl is essentially an alias for ifnull.
- Rewrite nvl2(x, y, z) to if(is_not_null(x), y, z)
- Add some tests

- Fixes #14381

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14421)

### CODE REVIEW

File: src/query/ast/src/parser/token.rs [hunk 1]
```diff
@@ -1093,6 +1093,10 @@ pub enum TokenKind {
     RANDOM,
     #[token("IFNULL", ignore(ascii_case))]
     IFNULL,
+    #[token("NVL", ignore(ascii_case))]
```
Comment (andylokandy) [inline src/query/ast/src/parser/token.rs:1096]: why need to define token?

Comment (PsiACE) [reply src/query/ast/src/parser/token.rs:1096]: Sorry I got some things wrong, I will remove them, they are not in the keywords.

File: src/query/sql/src/planner/semantic/type_check.rs [hunk 2]
```diff
@@ -2503,6 +2505,37 @@ impl<'a> TypeChecker<'a> {
...
+                    self.resolve_function(span, "if", vec![], &[
+                        &Expr::IsNull {
+                            span,
+                            expr: Box::new(arg_x.clone()),
+                            not: false,
+                        },
+                        arg_y,
+                        arg_x,
+                    ])
+                    .await,
```
Comment (andylokandy) [inline src/query/sql/src/planner/semantic/type_check.rs:2521]:
```suggestion
                    self.resolve_function(span, "if", vec![], &[
                        &Expr::IsNull {
                            span,
                            expr: Box::new(arg_x.clone()),
                            not: true,
                        },
                        arg_x,
                        arg_y,
                    ])
                    .await,
```

Comment (andylokandy) [reply src/query/sql/src/planner/semantic/type_check.rs:2521]: function is_not_null is better in performance than is_null

**Review by andylokandy — APPROVED** [review]
rest LGTM

**Comment by TCeason** [comment]
Maybe a flaky in https://github.com/datafuselabs/databend/actions/runs/7619610934/job/20753225934#step:6:1

**Review by TCeason — APPROVED** [review]

**Pull request merged**